File: Reverse Integer/reverse-integer.php
<?php
/**
 * Created by PhpStorm.
 * User: kovalevcon
 * Date: 05.10.18
 * Time: 15:04
 */

/**
 * Reverse integer function
 *
 * @param int $x
 * @return int
 */
function reverse(int $x): int
{
    $minInt = pow(-2, 31);
    $maxInt = pow(2, 31) - 1;

    if ($x == 0 || $x > $maxInt || $x < $minInt) {
        return 0;
    }

    $rev = (int)strrev((string)abs($x)) * ($x / abs($x));

    return $rev < $maxInt && $rev > $minInt ? $rev : 0;
}

echo "<pre>reverse(-123)\r\n";
echo reverse(-123); // -321
echo "</pre>";

echo "<pre>reverse(1534236469)\r\n";
echo reverse(1534236469); // Over max int, then result 0
echo "</pre>";

File: Reverse Integer/reverse-integer.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        min_int = -2**31
        max_int = 2**31 - 1

        if x == 0 or max_int < x < min_int:
            return 0

        rev = int(str(abs(x))[::-1]) * int((x / abs(x)))
        return rev if max_int > rev > min_int else 0


solution = Solution()
print(solution.reverse(-123))  # -321
print(solution.reverse(1534236469))  # Over max int, then result 0
